doc/source/ray-core/objects/limiting-memory-usage.rst

```diff
@@ -0,0 +1,274 @@
+    Consider provisioning more memory on this node or reducing task parallelism by requesting more CPUs per task. To adjust the eviction threshold, set the environment variable `RAY_memory_usage_threshold_fraction` when starting Ray. To disable worker eviction, set the environment variable `RAY_memory_monitor_interval_ms` to zero.
+
+Verify the task was indeed executed twice via ``task_oom_retry``:
+
+.. code-block:: bash
+
+    grep -r "retries left" /tmp/ray/session_latest/logs/
+
+    /tmp/ray/session_latest/logs/python-core-driver-01000000ffffffffffffffffffffffffffffffffffffffffffffffff_60002.log:[2022-10-12 16:14:07,723 I 60002 60031] task_manager.cc:458: task c8ef45ccd0112571ffffffffffffffffffffffff01000000 retries left: 3, oom retries left: 1, task failed due to oom: 1
+    
+    /tmp/ray/session_latest/logs/python-core-driver-01000000ffffffffffffffffffffffffffffffffffffffffffffffff_60002.log:[2022-10-12 16:14:18,843 I 60002 60031] task_manager.cc:458: task c8ef45ccd0112571ffffffffffffffffffffffff01000000 retries left: 3, oom retries left: 0, task failed due to oom: 1
+    
+    /tmp/ray/session_latest/logs/python-core-driver-01000000ffffffffffffffffffffffffffffffffffffffffffffffff_60002.log:[2022-10-12 16:14:18,843 I 60002 60031] task_manager.cc:466: No retries left for task c8ef45ccd0112571ffffffffffffffffffffffff01000000, not going to resubmit.
+
+.. note::
+
+    task retries are executed immediately. If there is a long running process it is possible for a task to keep retrying and fail and exhaust the oom retries. Consider increasing ``RAY_task_oom_retries``, :ref:`increase num_cpus per task to reduce the parallelism <limiting-memory-usage-increase-num-cpus>`, or :ref:`estimate the memory requirement and use memory aware scheduling <limiting-memory-usage-memory-scheduling>`.
+
+.. note::
+
+    actors by default are non-retriable since :ref:`max_restarts <actor-fault-tolerance>` defaults to 0, therefore tasks are preferred to actors when it comes to what gets killed first. Actor right now doesn't use ``RAY_task_oom_retries`` and instead uses :ref:`max_restarts <actor-fault-tolerance>` when it is killed by the memory monitor. This is to be changed very soon.
+
+.. _limiting-memory-usage-worker-killing-policy:
+
+Worker killing policy
+---------------------
+
+The raylet prioritizes killing task that is retriable, i.e. when ``max_retries`` or ``max_restarts`` is > 0. This is done to minimize workload failure. It then looks for the last one submitted and kills that worker process. It selects and kills one process at a time regardless of how frequent the monitor runs as determined by ``RAY_memory_monitor_interval_ms``.
+
+Let's first start ray and specify the memory threshold.
```

Comment:
Again, not sure we need the examples here since I think the text is straightforward enough on its own. Maybe could add to the design pattern examples, though, and link to this page?

Reply:
For worker killing policy: add a dropdown for example

num_cpus & memory scheduling : consolidate to "addressing memory issues" section and refer to patterns page for resource scheduling